### ISSUE TITLE
firehol 3.1.5: any firehol command results in error message

### ISSUE THREAD

**Issue by xeji** [NixOS/nixpkgs#35114]
## Issue description

The `firehol` firewall program in package firehol is unusable, always exits with an error message (see steps to reproduce).

Cause: On startup, `firehol` tries to create the config file `${pkgs.firehol}/etc/firehol/firehol-defaults.conf` and fails since the nix store is read-only.

How to fix: Make `firehol` look for config files in `/etc/firehol` rather than `${pkgs.firehol}/etc/firehol`. This makes sense anyway since a fixed bundled firewall config would not be useful.

I already implemented the fix. PR follows after testing.

### Steps to reproduce

1. install firehol with `onlyQOS=false` to enable the firehol program (the firehol package doesn't install the firehol program by default -  doesn't make sense to me, but that's a separate issue)
```
# nix-env -i -f '<nixpkgs>' -E 'f: (f {}).firehol.override {onlyQOS=false;}'
```

2. running _any_ firehol command results in this error message:
```
# firehol start
/root/.nix-profile/bin/firehol: line 1410: /nix/store/qc7zdhd0rs0cb0yhaam54rdjrz55zmhz-firehol-3.1.5/etc/firehol/firehol-defaults.conf: Read-only file system
```

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.9.81, NixOS, 18.03pre127921.9b25b9347d9 (Impala)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.16`
 - channels(root): `"nixos-18.03pre127921.9b25b9347d9"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

**Comment by xeji** [comment]
After fixing this, `firehol start` fails with a different error message. I discovered another bug caused by an unquoted command path in the `firehol` bash script. Fixed that as well. Working on a PR.

**Comment by xeji** [comment]
fixed in #35121
